icc-x-api/icc-message-x-api.ts

```diff
@@ -1043,120 +1060,124 @@ export class IccMessageXApi extends iccMessageApi {
     const year = moment().year()
 
     return getFederaton(invoices, this.insuranceApi).then(fed => {
-      const fedCode = fed.code === "306" ? "300" : fed.code
-      const prefix = `efact:${hcp.id}:${year}:${fedCode}:`
-      return this.entityReferenceApi
-        .getLatest(prefix)
-        .then((er: EntityReference) => {
-          let nextSeqNumber =
-            er && er.id && er.id!.startsWith(prefix)
-              ? (Number(er.id!.split(":").pop()) || 0) + 1
-              : 1
-          return this.entityReferenceApi.createEntityReference(
-            new EntityReference({
-              id: prefix + _.padStart("" + (nextSeqNumber % 1000000000), 9, "0"),
-              docId: uuid
-            })
-          )
-        })
-        .then(er =>
-          toInvoiceBatch(
-            invoices,
-            hcp,
-            fullBase36,
-            er && er.id ? Number(er.id.substr(prefix.length)) % 1000 : 0,
-            smallBase36,
-            this.insuranceApi,
-            this.invoiceXApi,
-            this
+      return (prefixer
+        ? prefixer(fed, hcp.id!)
+        : Promise.resolve(`efact:${hcp.id}:${year}:${fed.code === "306" ? "300" : fed.code}:`)
```

Comment:
https://github.com/antoinepairet/icc-api/pull/113#discussion_r280305421

@nicolas-b12 @phil1618 The regression (absence of year in the prefix) seems to have been fixed